language-design.md

```diff
@@ -34,6 +48,8 @@ How might the themes of _Growing a Language_ relate to ideas from the Fowler rea
 
 **Response**
 
+Fowler talks about how DSL's should be very simple and therefore easier to learn. This is similar to how Steele's argument that overbuilding a new language will scare away new users because it will be hard to learn. DSL's should be on the extreme end of simplicity though. General purpose languages that are grouping should be a mix between simple and complex. Steele's article explains why DSL's can be important for programming languages. If there is a specific niche use for a general purpose language that someone wants to add more functionality to, then a DSL is the best option because you wouldn't want to subject all the users of the general purpose language to learn them if only a few people are going to use it. DSL's allow programmers to add more advanced features to a language while also keeping the language in general easy to learn for everyone else. The idea of the "cost of building" a DSL could also be applied to any new programming language. If there is no interest in the language then it is probably not worth building. If a language would be useful to programmers, then it will generate a lot of interest and support and be much easier to build, but if nobody is interested it will be very hard to build and then nobody would use it anyway, so it is not worth it. Fowler also talks about how a DSL is "something that's evolving, not finished". This is similar to how a new programming language should be built. It should be designed so that it can be continued on in the future. 
```

Comment:
I definitely agree with you here, especially when you commented on the benefits of a DSL when a specific niche needs to be filled. In my response I made very similar comments, but your response helped me think about the fact that all users wouldn't want to be subjected to a specific use of a general purpose language when only a few people will end up using it. Keeping programming languages simple and easy to learn should definitely be a priority, and adding advance features through the use of DSL's is a great way to maintain this goal. I also liked how you talked about how if a programming language is something people are interested in it will be easier to build. This is especially true because languages are able to evolve faster when more people use them.